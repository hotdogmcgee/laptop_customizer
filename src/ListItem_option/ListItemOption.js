import React from 'react'

class ListItem_option extends React.Component {
    constructor(props) {
        super(props)
    }

    //destructure inside render
    render() {

        const {
            item: {
                  name,
                  cost,
            }
       } = this.props
       

        // const selectedClass = name === this.state.selected[key].name ? 'feature__selected' : '';
        // const featureClass = 'feature__option ' + selectedClass;
        return (
            <li className="feature__item">
            <div 
                // className={featureClass}
                onClick={this.props.updateFeature()}
            >
                { name }
                ({ new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD'})
                    .format(cost) })
            </div>
            </li>
        )
      
    }
}

export default ListItem_option