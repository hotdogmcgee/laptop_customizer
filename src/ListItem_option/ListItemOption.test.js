import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ListItemOption from './ListItemOption.js'


describe('ListItemOption component', () => {
  it('renders without crashing', () => {
    const testObj = {
        name: "hello",
        cost: 45
    }
    const div = document.createElement('div');
    ReactDOM.render(<ListItemOption item={testObj} updateFeature={() => {return 1}}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const testObj = {
        name: "hello",
        cost: 45,
    }
    const tree = renderer
      .create(<ListItemOption item={testObj} updateFeature={() => {return 1}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});