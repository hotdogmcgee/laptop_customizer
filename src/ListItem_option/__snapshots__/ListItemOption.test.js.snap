// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemOption component renders the UI as expected 1`] = `
<li
  className="feature__item"
>
  <div
    onClick={1}
  >
    hello
    (
    $45.00
    )
  </div>
</li>
`;
